src/igraph/__init__.py

```diff
@@ -3205,6 +3205,71 @@ def TupleList(
         # Construct the graph
         return cls(n, edge_list, directed, {}, vertex_attributes, edge_attributes)
 
+    #####################################################
+    # Constructor for dict-like representation of graphs
+
+    @classmethod
+    def SequenceDict(
+        cls,
+        edges,
+        directed=False,
+        vertex_name_attr="name",
```

Comment:
I think the vertex name should always be called `name`, otherwise `igraph` would not recognize it as a proper "name" (i.e. including the indexing on the name), right?

Reply:
No, I think it's okay this way; there could be perfectly valid reasons for trying to construct a graph where the `name` attribute would be provided later by the user and the original names are stored in a different attribute (e.g., `original_name`).